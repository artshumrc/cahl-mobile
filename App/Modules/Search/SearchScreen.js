import React from 'react';
import { ScrollView, View, Text, TextInput, TouchableOpacity } from 'react-native';
import I18n from 'react-native-i18n';

// styles
import styles from './SearchScreenStyles.js';

// translations
import en from '../../i18n/languages/english.json';
import fr from '../../i18n/languages/fr.json';

I18n.translations = {
  en,
  fr,
};

class SearchScreen extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      exhibitNumber: '',
    };
  }

  handleSubmit() {
    if (this.state.exhibitNumber != '') {
      console.log('Looking up exhibit #', this.state.exhibitNumber);
    } else {
      console.log('Enter an exhibit number');
    }
  }

  render() {
    const { exhibitNumber } = this.state;
    const buttonStyle = exhibitNumber !== '' ? styles.button : styles.buttonNotReady;
    const buttonTextStyle = exhibitNumber !== '' ? styles.buttonText : styles.buttonTextNotReady;
    return (
      <View style={styles.container}>
        <Text style={styles.instructionText}>{I18n.t('searchItem')}</Text>
        <TextInput
          value={exhibitNumber}
          onChange={exhibitNumber => this.setState({ exhibitNumber })}
          keyboardType="numeric"
          style={styles.textInput}
          autoFocus
          textAlign={'center'}
          returnKeyType="search"
          maxLength={3}
        />
        <TouchableOpacity
          style={buttonStyle}
          onPress={() => this.handleSubmit}
        >
          <Text style={buttonTextStyle}>{I18n.t('viewItem')}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export default SearchScreen;
