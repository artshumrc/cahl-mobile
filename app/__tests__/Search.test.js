import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SearchScreen from '../modules/Search';

it('renders correctly', () => {
  const tree = renderer.create(
    <SearchScreen />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
