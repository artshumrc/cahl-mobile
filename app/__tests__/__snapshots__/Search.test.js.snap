// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#999999",
        "fontFamily": "AvenirNext-regular",
        "fontSize": 20,
      }
    }
  >
    Enter an item number
  </Text>
  <TextInput
    clearTextOnFocus={true}
    keyboardType="number-pad"
    onChange={[Function]}
    style={
      Object {
        "height": 10,
        "width": 50,
      }
    }
    value="- - -"
  />
</View>
`;
